crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -54,7 +54,7 @@ fn process_abi_with_input(
         match &return_param.1 {
             AbiType::Array { length, .. } => *length as u32,
             AbiType::Integer { .. } | AbiType::Field(_) => 1,
-            AbiType::Struct { num_fields, .. } => *num_fields as u32,
+            AbiType::Struct { fields, .. } => (*fields).len() as u32,
```

Comment:
Dereferencing fields shouldn't be necessary here since `.` uses the Deref trait. Were you getting an error otherwise?

Reply:
I think my VS code was caching an old error or something. Good catch, I am no longer dereferencing here and no error